pkg/upload/react-native-android.go

```diff
@@ -105,41 +120,46 @@ func ProcessReactNativeAndroid(apiKey string, appManifestPath string, bundlePath
 			if utils.FileExists(appManifestPathExpected) {
 				appManifestPath = appManifestPathExpected
 				log.Info("Found app manifest at: " + appManifestPath)
+			} else {
+				appManifestPath = ""
 			}
 		}
 
-		if appManifestPath != "" && (apiKey == "" || version == "" || versionCode == "") {
-
-			manifestData, err := android.ParseAndroidManifestXML(appManifestPath)
+		if apiKey == "" || versionName == "" || versionCode == "" {
+			if appManifestPath != "" {
```

Comment:
Was this causing an issue? I believe this was the intended approach - we read the manifest if any of the data is missing and only if the manifest is found. Otherwise we allow `BuildReactNativeAndroidUploadOptions` to generate the error.